File: Back-End/ItsLib.DAL/Data/ProductUser.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations.Schema;

namespace ItsLib.DAL.Data
{
    [PrimaryKey(nameof(UserId), nameof(ProductId))]
    public class ProductUser
    {
        [ForeignKey("User")]
        public string UserId { get; set; }

        [ForeignKey("Product")]
        public Guid ProductId { get; set; }
        public bool InWishList { get; set; }
        public bool IsUsed { get; set; }
        public bool IsDisabled { get; set; }
        public string? Review { get; set; }
        public string? ReviewTitle { get; set; }
        public DateTime DateAdded { get; set; }
        public DateTime LastModifiedDate { get; set; }
        public User User { get; set; }
        public Product Product { get; set; }
    }
}

File: Front-End/src/app/interfaces/product.user.ts
import { Product } from "./product";
import { UserOp } from "./user.op";

export interface ProductUser {
    userId: string,
    productId: string,
    inWishList: boolean,
    isUsed: boolean,
    review: string | null,
    reviewTitle: string | null,
    isDisabled: boolean,
    dateAdded: Date,
    lastModifiedDate: Date,
    user: UserOp | undefined,
    product: Product | undefined
}
